cmd/virt-v2v-monitor/virt-v2v-monitor.go

```diff
@@ -108,6 +204,15 @@ func main() {
 			progress, _ = strconv.ParseUint(string(match[1]), 10, 0)
 			klog.Infof("Progress update, completed %d %%", progress)
 			err = updateProgress(progressCounter, diskNumber, progress)
+		} else if strings.Contains(string(line), XML_CONF) {
+			klog.Infof("Strating reading xml configuratoin")
+			inXML = true
+		} else if inXML {
+			xmlData.WriteString(string(line) + "\n")
+			if strings.Contains(string(line), XML_CONF_END) {
+				klog.Infof("Finished reading xml configuratoin")
+				inXML = false
+			}
```

Comment:
i find this part strange. we need to iterate this twice? once to open the XML and once to end it?

Reply:
nope, since we using virt-v2v with `-v -x` parameters set (for logging and debugging) the xml file generated is not being separately created but rather printed to the logs. so we need somehow read it and save it to a variable, since the metrics code already doing this for the progress I've decided to use it for reading this configuration as well to reduce code dups. so its basically checking for the xml header and once it detected writes all the xml lines into var until its encounters the last line of the xml where its always the same.

Reply:
wouldn't it be nicer, if possible, to make virt-v2v print it to a file, then just read the file?